cypress/integration/ColorPicker.spec.ts

```diff
@@ -129,48 +125,52 @@ describe('ColorPicker', () => {
       });
 
       context('when a swatch is clicked', () => {
-        const color = '8660d1';
+        const color = '#ff5347';
+
         beforeEach(() => {
-          cy.get(`.wdc-color-picker--color-${color}`).click();
-          cy.get('button').click();
+          getSwatch(color).click();
+          getOpenButton().click();
         });
 
         it('should have check icon', () => {
-          cy.get(`.wdc-color-picker--color-${color}`)
+          getSwatch(color)
             .find('.wd-icon')
             .should('exist');
         });
-      });
 
-      context('when color reset is clicked', () => {
-        beforeEach(() => {
-          cy.get(`.wdc-color-picker--color-8660d1`).click();
-          cy.get('button').click();
-          cy.get('[data-testid="color-picker-reset"]').click();
-        });
+        context('when color reset is clicked', () => {
+          const color = '#ff5347';
+          let resetColor: string | undefined;
 
-        it('should set the color picker value to the reset color', () => {
-          cy.get('button').click();
-          cy.get(`.wdc-color-picker--color-0875e1`)
-            .find('.wd-icon')
-            .should('exist');
+          beforeEach(() => {
+            getResetButton()
+              .then($button => {
+                resetColor = $button.find('[color]').attr('color');
+              })
+              .click();
```

Comment:
Not sure if there's a better way to do this. I want to store the value of the reset color from the yielded button before I click on it. @NicholasBoll any thoughts here?

Reply:
I see this type of thought a lot for testing, but I tend to disagree. I think if the data is known, we should assert against that known data, removing the need for a variable. A variable means there is logic inside the tests and that logic is often itself untested (although often implicitly tested).

This comes down to the idea that we are making specifications for our components and patterns. A variable is more flexible - the reset color can change without the test changing. But our specifications are about what we expect of our code. If the reset color changes, so has our specifications of the reset color.

I believe variables should almost always be avoided. Most of the use-cases with variables involve a lack of control of the initial state of a system (like seeding database data and getting guids for domain objects, or not being able to accurately seed data and dealing with "random" initial data). I don't think this variable falls into a lack of control of initial state.

A way around a variable in this case is that the reset color value is hard-coded. I don't think the point of a good test is to never fail, but fail only when we expect it to fail and it being easy to understand why and how it fails.